src/components/PostDetail.js

```diff
@@ -55,45 +51,78 @@ const styles = StyleSheet.create({
 const PostDetail = React.createClass({
   propTypes: {
     post: postPropTypes.isRequired,
+    navigator: postPropTypes.isRequired,
   },
 
-  render: function() {
+  getHTML: function() {
     const post = this.props.post;
-    let caption;
-    if (post.images.length > 0 && post.images[0].caption !== '') {
-      caption = (
-        <Text style={styles.caption}>{post.images[0].caption}</Text>
-      );
-    }
-    let byline;
-    if (post.authors) {
-      byline = (
-        <Text style={styles.byline}>By {post.authors.join(', ')}</Text>
-      );
+
+    const caption = (post.images.length > 0 && post.images[0].caption !== '') ? post.images[0].caption : '';
+    const author = (post.authors) ? `<p id='byline'>By ${post.authors.join(', ')} </p>` : '';
+    const image = (post.images.length > 0) ? `<img id='image' src=' ${post.images[0].previewUrl} '>` : '';
+    const disqusHTML = `
+      <!DOCTYPE html>
+      <div id='disqus_thread'></div>
+      <script type='text/javascript'>
+        var disqus_url = '${post.url}';
+        (function(){ var dsq = document.createElement('script');
+        dsq.type = 'text/javascript';
+        dsq.async = true;
+        dsq.src = 'http://dukechronicle.disqus.com/embed.js';
+        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq); })();
+      </script>
+    `;
+
+    return `
+      <!DOCTYPE html>
+      <html>
+      <head>
+      <style> ${innerStyles} </style>
+          <p id='title'> ${post.title} </p>
+          <p id='caption'> ${caption} </p>
+          <p id='byline'> ${author} </p>
+          ${image}
+          ${post.body}
+          ${disqusHTML}
+      </head>
+      </html>
+    `;
+  },
+
+  redirect: function(navState) {
+    if (navState.loading === true) {
+      return;
     }
-    let image;
-    if (post.images.length > 0) {
-      image = (
-        <Image
-          style={styles.image}
-          source={{uri: post.images[0].previewUrl}}
-        />
-      );
+    const url = navState.url;
+    const successes = [
+      'disqus.com/next/login-success',
+      'disqus.com/_ax/google/complete',
+      'disqus.com/_ax/twitter/complete',
+      'disqus.com/_ax/facebook/complete',
+    ];
+
+    if ((url.split('.com'))[0].contains('disqus') && _.some(successes, (success) => {return url.contains(success); })) {
+      this.props.navigator.push({
+        title: this.props.title,
```

Comment:
`this.props.post.title` ? This can probably just be changed to `''` though, because in the new navigator bar post screens don't show titles.